src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject_IIntellisenseBuildTarget.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Build.Execution;
+using Microsoft.Build.Logging;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    internal partial class AbstractProject : IIntellisenseBuildTarget
+    {
+        private static readonly object s_diagnosticKey = new object();
+        private static readonly FieldInfo s_fieldInfo = typeof(BuildManager).GetField("_buildParameters", BindingFlags.GetField | BindingFlags.Instance | BindingFlags.NonPublic);
```

Comment:
@dotnet/roslyn-ide @dotnet/roslyn-analysis well, it depends. all your saying is for us, not for users, if people agree, then sure. but for me, if I am user, I dont care all those, all I want is whether I get better information to act on. and this does give that information. 

the bug associated with this fix has 60 projects intellisense build failure, and without that information, I had to go through 60 files to find issues on a project I am interested in. this give point to right log right away.